core/unix/loader.c

```diff
@@ -732,6 +733,11 @@ privload_search_rpath(privmod_t *mod, bool runpath, const char *name,
             const char *list = strtab + dyn->d_un.d_val;
             const char *sep, *origin;
             size_t len;
+            /* XXX: The list is already split into it's elements. Currently, at least on
+             * the systems observed when writing this comment, this loops spills over
+             * across the actual rpath or runpath list, enclosed by '[' ']', and iterates
```

Comment:
WDYM "enclosed by []"?  The string in the binary does not have brackets:
```
  RUNPATH              $ORIGIN/../../lib64/debug
```

I also don't see any overrun: it is null-terminated in the case in front of me.

Reply:
In the toolchain I am looking at, the string is

```
0x000000000000000f (RPATH)              Library rpath: [$ORIGIN/../../_solib__third_Uparty_Scrosstool_Sv18_Sstable_Ccc-compiler-k8-llvm-libcxx/:$ORIGIN/_solib__third_Uparty_Scrosstool_Sv18_Sstable_Ccc-compiler-k8-llvm-libcxx/]
```

Maybe this is new? I debugged and see it go past the string into DT_NEEDED area.

Reply:
The brackets are added by readelf.  They are not part of the string.

Reply:
It does look look it goes too far: it should be list+=len for the non-: case, right?  The len+1 is for the : case only.

Reply:
Ok, you're right. Still seeing the overrun, checking why...

Reply:
Yes, [that's it](https://github.com/DynamoRIO/dynamorio/pull/3854#discussion_r329231023).
I'll include a fix